### PR TITLE
map bytes to Uint8array

### PR DESCRIPTION
**Pull request by mschneider**
Uint8Array is a js native representation for an array of unsigned bytes, the current Array<number> mapping has bad performance characteristics, especially when working with large buffers. I'd rather leave the Array conversion to the user than making it part of the generator.

### CODE REVIEW

**Comment by mschneider** [comment]
I ran yarn build and yarn test locally to confirm this doesn't break anything, but noticed that I had a bunch of changed files not sure why. LMK if i should reset some of them, or if you want to merge them.

File: src/common.ts
Comment (kklas) [inline src/common.ts:664]: Shouldn't this be `Uint8Array.from(<>)`?

File: tests/test.ts
Comment (kklas) [inline tests/test.ts:102]: Could we use `Uint8Array.from(...)` in this filed instead of Buffer? I know Buffer is a subclass of Uint8Array but it's a bit confusing.

**Review by kklas — CHANGES_REQUESTED** [review]
Yes you should include files generated by yarn build. These include the examples generated with the new code as well as the test example-program idl and expected codegen.
The docs fields in `tests/example-program-gen/idl.json` are removed in your commit which probably means you've ran `anchor build` manually and that's why the generated example-program code doesn't include docs. With `yarn build` it should use the nightly version of anchor which should include those comments.

Run `yarn build` and include all files. Otherwise the CI tests will fail. Also please update CHANGELOG.md.

File: package.json
Comment (mschneider) [reply package.json:54]: Need to move rimraf and tsc to dependencies from devDependencies else they can't be executed during postinstall stage. It allows to add anchor-client-gen as npm dependency to a project rather then requiring people to install it globally, possibly causing versioning projects between multiple projects.

**Comment by mschneider** [comment]
I ran yarn build, but the npm package doesn't work on my platform (darwin/arm64):

```
Only x86_64 / Linux distributed in NPM package right now.
Trying globally installed anchor.
```

File: tests/test.ts
Comment (mschneider) [reply tests/test.ts:102]: yes

File: src/common.ts
Comment (mschneider) [reply src/common.ts:664]: yes

**Comment by kklas** [comment]
Ah that makes sense. It ran your globally installed version of anchor which is < anchor 0.25 and that's why the docs aren't generated.
No worries, I'll build it before merging.

File: tests/test.ts
Comment (mschneider) [reply tests/test.ts:102]: The bytes field is actually a buffer, and the comparison fails if the right hand side is not a buffer either.

Comment (kklas) [reply tests/test.ts:102]: Hmm, does the borsh layout decode return a Buffer instead of Uint8Array in that case?

Comment (mschneider) [reply tests/test.ts:102]: Yeah borsh seems to be pretty much Buffer based.

File: package.json
Comment (kklas) [reply package.json:54]: Hmm now that I think about it I don't think we need this (the ability to link the package via git branch and the postinstall script). If you want to get the unreleased features there's the beta package, I can update it for you if you want to use the latest commits. And if you want to use your local package you can always yarn / npm link.
In any case I think this is out of scope for this PR. Can you leave the package.json as it was for now and let's have this discussion on another PR / issue?

Comment (mschneider) [reply package.json:54]: Well i needed it to continue with what i wanted to work on as I'm not the maintainer of this repo. I don't think yarn link is a workable solution, when working with two other people on the same project, as it doesn't update lock files and hence produces inconsistent git commits. Reverted the commit anyways.

**Comment by kklas** [comment]
Alright, looks good overall! I'll finish it up from here with the build. Thanks for the PR!

**Review by kklas — APPROVED** [review]

**Pull request merged**
